### ISSUE TITLE
[features request] Accept user messages, #general takes you to general, and a UI rework, emoji picker.

### ISSUE THREAD

**Issue by tawnylure** [uowuo/abaddon#224]
Here are some features that would increase the QoL in Abaddon and make it easier to use.

1 Just like in discord if someone does like #general it should take you to the general chat if you click on it, currently its displayed as only text though.

2  Accepting user messages, pretty neeed feature. As you are unable to accept new messages. Unless you right click on their profile, and go to dm's which is time consuming and you need to know their profile.

3. A general UI rework, something kinda like gtkcord4, its UI is like discords, you click on a server and see a list of channels, instead of the mess that the current UI is, displaying channels in the same area as servers is complicated and messy.

4. Emoji picker, its kinda annoying having to remember your faviorate emojis such as :rage: and :smile:, or just having to type it out in general to get the same result of a emoji picker gui, Kinda like discord.

5. Better image viewing / previews. Being able to view something like a Image / YouTube video should be a feature aswell, instead of redirecting to a browser for a enlarged image view you could have a pop up like discord. or being able to view a YouTube video in the app itself.
Could also make image previews for image sharing sites like imgur larger instead of the tiny window they are currently.

6. Proper support for threads, they legit just dont show up on the client. Kinda a killer for support forums etc.

All these changes would not only be a good QoL update but add more functionality to the app and make the official discord client less and less appealing.

Now of course I also don't expect any of these features to be added, its more of a "wishlist" of things that would make the user experience 100x better. Y'all doing great work with this project, its nice to see a non electron version of discord that has pretty good functionality. Keep up the good work as always.

**Comment by tawnylure** [comment]
Im open to feedback on these.

**Comment by tawnylure** [comment]
Edit - Upon further inspection their is indeed a emote wheel, though its only accessible via a right click on the text bar.

Perhaps a icon to open this would be beneficial.

**Comment by ouwou** [comment]
1. should already be a thing. channel mentions should work about the same as they do in the official client
2. ill look into that
3. i personally prefer the tree view channel list but maybe ill add a setting to use a different UI for it. its certainly possible
4. the emoji picker is built into the gui toolkit and isnt that good. i do need to make a custom one for sure
5. youtube probably wont happen since that effectively requires embedding a web browser. ive had the image expansion idea suggested before and its a good idea
6. threads should work mostly fine. the client cant know about them until you load a channel thats part of the same server though (this is a limitation of discord's api). threads you are considered a member of should show up in the channel list and you should be able to see the rest under the View > Threads menu. unless there is some other problem then let me know

**Comment by tawnylure** [comment]
> 1. should already be a thing. channel mentions should work about the same as they do in the official client
> 
>     2. ill look into that
> 
>     3. i personally prefer the tree view channel list but maybe ill add a setting to use a different UI for it. its certainly possible
> 
>     4. the emoji picker is built into the gui toolkit and isnt that good. i do need to make a custom one for sure
> 
>     5. youtube probably wont happen since that effectively requires embedding a web browser. ive had the image expansion idea suggested before and its a good idea
> 
>     6. threads should work mostly fine. the client cant know about them until you load a channel thats part of the same server though (this is a limitation of discord's api). threads you are considered a member of should show up in the channel list and you should be able to see the rest under the View > Threads menu. unless there is some other problem then let me know

Hm, good to know. 
And channel mentions do not seem to work correctly. Anytime someone mentions a channel it isnt displayed as a channel mention. So might wanna wanna take a look into it.

**Comment by tawnylure** [comment]
And dedicated thread channels do not work, im talking about the forum feature I'll link a screen shot of what im talking about.
Like this channel just doen't appear on abaddon - https://i.imgur.com/NCtHwWJ @ouwou

**Comment by ouwou** [comment]
forums ill get to eventually. ive just never had to use them so they arent implemented xd
as for the channel mentions, do they show up bold and parsed (i.e. #channel and not <#1999bunchofnumbers>)? what happens if you try to send a channel mention yourself?

**Comment by tawnylure** [comment]
> forums ill get to eventually. ive just never had to use them so they arent implemented xd as for the channel mentions, do they show up bold and parsed (i.e. #channel and not <#1999bunchofnumbers>)? what happens if you try to send a channel mention yourself?

Upon looking further, it seems clicking on it is a bit buggy, in some areas it works, in others it doesnt. It works best if you click in the middle area for a channel mention, idk why but its weird.

**Comment by gschurck** [comment]
Also highly interested by a better UI, closer to gtkcord4 with different columns for servers and channels.

**Comment by tawnylure** [comment]
> Also highly interested by a better UI, closer to gtkcord4 with different columns for servers and channels.

Yeah same, though I don't want it looking exactly like gtkcord4, I personally like the STYLE of it, the layout is kinda dog atleast for servers. I hope they add a option sooner rather then later to have the channels popout into another area then the server browser.

Otherwise this projects honestly really good, a *almost* fully featured gtk based discord client?? like shucks if it had forums and a better organized UI, i would recommend this to everyone
